ouroboros-network/src/Ouroboros/Network/Diffusion.hs

```diff
@@ -65,6 +69,26 @@ import           Ouroboros.Network.Subscription.Dns
 import           Ouroboros.Network.Subscription.Worker (LocalAddresses (..))
 import           Ouroboros.Network.Tracers
 
+-- | Socket file descriptor.
+--
+newtype SocketFD = SocketFD { getSocketFD :: CInt } deriving Eq
+
+instance Show SocketFD where
+  show fd = "<socket: " ++ show (getSocketFD fd) ++ ">"
+
+data DiffusionInitializationTracer
+  = RunServer
+  | RunLocalServer
```

Comment:
I would add `SockAddr` and `LocalAddress` on both `RunServer` and `RunLocalServer`. These are probably the only two messages that should be logged with `Notice` severity, all others are either `Info` or `Debug` messages so it makes sense to show the addresses on both of them.

Reply:
I wasn't able to get a `SockAddr` or `LocalAddress` for these points of logging, so I used `FileDescriptor` instead.  Please advise.

Reply:
For `runServer` you can do this:
```
          addr <- case address of
               Left sd -> getLocalName sn sd
               Right addr -> do
                 traceWith dtDiffusionInitializationTracer $ ConfiguringServerSocket addr
                 Snocket.bind sn sd addr
                 return addr
                 Snocket.listen sn sd

          traceWith dtDiffusionInitializationTracer (RunServer addr)
```

Reply:
[Snocket haddoc documentation](https://input-output-hk.github.io/ouroboros-network/ouroboros-network-framework/Ouroboros-Network-Snocket.html#t:Snocket) could help you find the solution (though it could be improved).